qiita_db/study.py

```diff
@@ -234,11 +254,15 @@ def get_info(cls, study_ids=None, info_cols=None):
             JOIN qiita.timeseries_type  USING (timeseries_type_id)
             LEFT JOIN (SELECT study_id, array_agg(pmid ORDER BY pmid) as
             pmid FROM qiita.study_pmid GROUP BY study_id) sp USING (study_id)
-            )""".format(search_cols)
+            JOIN qiita.study_portal USING (study_id)
+            JOIN qiita.portal_type USING (portal_type_id)
+            ) WHERE portal = '{1}'""".format(search_cols, qiita_config.portal)
         if study_ids is not None:
-            sql = "{0} WHERE study_id in ({1})".format(
-                sql, ','.join(str(s) for s in study_ids))
+            # Make sure portal has access to all studies asked for
+            cls._check_portal_access(study_ids)
```

Comment:
That is currently not implemented, and is going to take a lot more work. The checks are in the base class, so thats going to take some re-tooling.